blockdev/src/blockdev.rs

```diff
@@ -240,6 +303,123 @@ impl Drop for LoopbackDevice {
     }
 }
 
+/// Main function for the loopback cleanup helper process
+/// This function does not return - it either exits normally or via signal
+pub fn run_loopback_cleanup_helper(device_path: &str, parent_pid: u32) -> Result<()> {
+    use std::os::unix::process::CommandExt;
+    use std::process::Command;
+
+    // Check if we're running as a cleanup helper
+    if std::env::var("BOOTC_LOOPBACK_CLEANUP_HELPER").is_err() {
+        anyhow::bail!("This function should only be called as a cleanup helper");
+    }
+
+    // Close stdin, stdout, stderr and redirect to /dev/null
+    let null_fd = std::fs::File::open("/dev/null")?;
+    let null_fd = null_fd.as_raw_fd();
+    unsafe {
+        libc::dup2(null_fd, 0);
+        libc::dup2(null_fd, 1);
+        libc::dup2(null_fd, 2);
+    }
+    
+    // Set up death signal notification - we want to be notified when parent dies
+    unsafe {
```

Comment:
There's a safe version of this in https://docs.rs/rustix/latest/rustix/process/fn.set_parent_process_death_signal.html